ta4j-core/src/main/java/org/ta4j/core/BaseBar.java

```diff
@@ -57,6 +57,8 @@ this software and associated documentation files (the "Software"), to deal in
     private Num volume;
     /** Trade count */
     private int trades = 0;
+    /** Num function **/
+    private Function<Number, Num> numFunction;
```

Comment:
I have also thought about this `numFunction ` field in `BaseBar ` and I hesitated for a long time, but I think this solution is cleaner. It is also easier to make the `Num ` check in `TimeSeries ` if adding a new bar

Reply:
Like in `TimeSeries` the `Bar `interface should have a `function() ` getter for the `numFunction`. Then we can simplify the `Bar ` check in the `TimeSeries#addBar(Bar b)` implementation of `BaseTimeSeries`